corehq/apps/reports/static/reports/js/case_list.js

```diff
@@ -1,9 +1,26 @@
+hqImport("hqwebapp/js/initial_page_data").get('user_types')
 hqDefine("reports/js/case_list", ['jquery', 'analytix/js/kissmetrix'], function ($, kissAnalytics) {
     $(function () {
         if (document.location.href.match(/reports\/case_list/)) {
             $(document).on('click', 'td.case-name-link', function () {
                 kissAnalytics.track.event("Clicked Case Name in Case List Report");
             });
+            var user_types = hqImport("hqwebapp/js/initial_page_data").get('user_types');
+            $(document).on('click', 'button#apply-filters', function () {
```

Comment:
Minor fun fact: `#apply-filters` will perform better than `button#apply-filters`, because the former will use the vanilla js `getElementById`, which is very fast.

Reply:
yes, well, `querySelectorAll`. check out https://learn.jquery.com/performance/optimize-selectors/

Reply:
Good to know: https://github.com/dimagi/commcare-hq/pull/22079/commits/e3ba8050a26a8c3a04917954017e075402ed2134